webdriver-spec.html

```diff
@@ -4105,7 +4108,46 @@
 
  <li><p><a>Scroll into view</a> the <var>element</var>.
 
- <li><p class=issue>Interact with the element.
+ <li><p>Run the <a>focusing steps</a> for the <a>element</a>.
+
+ <li><p>Set the cursor position to after the last character in the element
```

Comment:
First of all, we probably want to reconcile sending key input from this command and from an action chain to a single definition.

Second, you only want to move the cursor to the end for this command, as the action API is supposed to enter keys wherever the current cursor is.

OTOH this algorithm needs to take into account not only `<input>` and `contenteditable` elements, but also `<textarea>`.  For `<input>` and `<textarea>` the following can be used:

```
let t = el.type;
if (t && (t == "text" || == "textarea")) {
  let len = el.value.length;
  el.setSelectionRange(len, len);
}
el.focus();
```

`contenteditable` shares some of the same challenges as when `document.designMode` is enabled.  I would suggest documenting the other things first, and file a bug to address editable elements later.